src/Hydrator/EloquentHydrator.php

```diff
@@ -253,12 +254,20 @@ protected function deserializeDate($value)
     /**
      * Is this resource key a date attribute?
      *
+     * @param $record
      * @param $resourceKey
      * @return bool
      */
-    protected function isDateAttribute($resourceKey)
+    protected function isDateAttribute($record, $resourceKey)
     {
-        return in_array($resourceKey, $this->dates, true);
+        if(is_null($this->dates))
+        {
+            return in_array(Str::snake($resourceKey), $record->getDates(), true);
+        }
+        else
```

Comment:
There is no need for the else here since you're returning early :)

```php
protected function isDateAttribute($record, $resourceKey)
{
    if (is_null($this->dates)) {
        return in_array(Str::snake($resourceKey), $record->getDates(), true);
    }

    return in_array($resourceKey, $this->dates, true);
}
```